### ISSUE TITLE
Add a warning when using a JSDoc deprecated method

### ISSUE THREAD

**Issue by louy** [palantir/tslint#825]
So this was previously suggested in Microsoft/TypeScript#390, but it seems like it should be in tslint, not typescript.

The idea is when a function which has the tag "[@deprecated](http://usejsdoc.org/tags-deprecated.html)" is used it should show a warning.
For example:

``` ts
interface IX {
  /**
   * @deprecated
   */
  function doSomething(): void;
}

class X implements IX {
  function doSomething(): void;

  /**
   * @deprecated
   */
  function doSomethingElse(): void;
}

const x = new X();
x.doSomething(); // Warning: IX.doSomething is deprecated
x.doSomethingElse(); // Warning: X.doSomethingElse is deprecated
```

Same thing should preferably apply to attributes. Getters & setters.

**Comment by adidahiya** [comment]
Interesting suggestion. This would involve telling TSLint about your typings files so that it could crawl JSDoc comments. We don't currently do this, and TSLint would have to treat typings different from sources (don't lint them since they probably don't have the same code style as your sources).

Related: #680, #692

**Comment by louy** [comment]
Yeah I agree. As to #680, that was actually one of the first frustrations I had with TSLint, coming from eslint.
I think we're stuck here until those two are closed then.

**Comment by danpantry** [comment]
Now that #680 is solved, what's the progress on this?

**Comment by adidahiya** [comment]
Open to PRs. I haven't yet looked at what language service APIs are available for reading JSDoc comments, but feel free to take a crack at it.
